### PR TITLE
Fix/user defined types parameterized with builtin types

### PR DESCRIPTION
**Pull request by danabr**

### CODE REVIEW

**Comment by danabr** [comment]
This is actually more parsing related than typing related. Should I move the test to the `mlfe` module instead?

**Comment by danabr** [comment]
Closes #20.

File: src/mlfe_parser.yrl [hunk 1]
```diff
@@ -96,14 +96,14 @@ type_import -> use module_fun:
 
 module_def -> module atom : {module, '$1'}.
 
-type_vars -> type_var : ['$1'].
-type_vars -> type_var type_vars : ['$1'|'$2'].
+type_vars -> type_expr : ['$1'].
```
Comment (j14159) [inline src/mlfe_parser.yrl]: Should we maybe change from `type_var` to something else like `type_expressions` or something along those lines?  Minor point but might alleviate confusion later :)

**Comment by j14159** [comment]
I'm OK with the test in either place as it's a parsing fix for a typing feature.  Wherever you're most comfortable with it.

**Comment by j14159** [comment]
Thanks for this!

**Pull request merged**